pkg/query/flamegraph_arrow.go

```diff
@@ -270,9 +281,106 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 	return record, fb.cumulative, fb.maxHeight + 1, 0, nil
 }
 
+type transpositions struct {
+	mappingIDIndicesData *array.Data
```

Comment:
nit: I think it would be nicer to unify these `*Data` and `*Indices` fields into a struct to halve the number of fields here and e.g. `Release` calls.

Reply:
Yes